### ISSUE TITLE
type hints in methods using PEP 604 result in horizontal line

### ISSUE THREAD

**Issue by cybercyber123** [pylint-dev/pylint#8671]
### Bug description

The following type hints cause unwanted horizontal lines, because of un-escaped pipe-symbols.
```
class FailingClass:
    def __init__(
        self,
        with_default: str | None = None,
    ) -> None:
        self._bar = with_default

    def the_failing_method(
        self,
        the_arg: str | None = None,
    ) -> None:
        pass
```

### Command used

```shell
pyreverse --filter ALL --output-directory output --output dot ./main.py
```

### Pylint output

```shell
digraph "classes" {
rankdir=BT
charset="utf-8"
"main.FailingClass" [color="black", fontcolor="black", label=<{FailingClass|_bar : Optional[str \| None]<br ALIGN="LEFT"/>|__init__(with_default: str | None): None<br ALIGN="LEFT"/><I>the_failing_method</I>(the_arg: str | None): None<br ALIGN="LEFT"/>}>, shape="record", style="solid"];
}
```

### Expected behavior

i would expect proper handling of those type hints.

### Pylint version

```shell
pylint==2.17.4
```

**Comment by DanielNoord** [comment]
I don't understand this issue. What do you mean with horizontal lines?

**Comment by cybercyber123** [comment]
![classes](https://github.com/pylint-dev/pylint/assets/123474936/30399720-35dc-4dee-b010-cd3310194ca7)

i am referring to the additional lines following init in provided svg
